site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,767 @@
+
+.. code-block:: java
+
+    @RequestConverter(MyRequestConverter.class)
+    public class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public HttpResponse hello(@RequestObject MyObject myObject) {
+            // MyRequestConverter will be used to convert a request.
+            // ...
+        }
+
+        @Post("/hola")
+        @RequestConverter(MySpanishRequestConverter.class)
+        public HttpResponse hola(@RequestObject MySpanishObject myObject) {
+            // MySpanishRequestConverter will be tried to convert a request first.
+            // MyRequestConverter will be used if MySpanishRequestConverter fell through.
+            // ...
+        }
+    }
+
+Armeria also provides built-in request converters such as, JacksonRequestConverterFunction_ for JSON
+documents, StringRequestConverterFunction_ for text contents and ByteArrayRequestConverterFunction_
+for binary contents. They will be applied after your request converters by default, so you can use these
+built-in converters by just putting `@RequestObject`_ annotation on the parameters which you want to convert.
+
+In some cases, `@RequestObject`_ annotation may have a request converter as its value.
+Assume that you have a Java class named ``MyRequest`` that it is usually able to be converted by
+``MyDefaultRequestConverter``. But what if there is only one method which has a parameter of ``MyRequest``
+that you have to convert it differently? In this case, you may specify a request converter with
+`@RequestObject`_ annotation. In the example, ``MySpecialRequestConverter`` will be used first for
+converting ``MyRequest``.
+
+.. code-block:: java
+
+    @RequestConverter(MyDefaultRequestConverter.class)
+    public class MyAnnotatedServiceClass {
+        @Post("/hello")
+        public HttpResponse hello(
+            @RequestObject(MySpecialRequestConverter.class) MyRequest myRequest) { ... }
+    }
+
+Converting a Java object to an HTTP response
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Every object returned by an annotated service method can be converted to an HTTP response message by
+response converters, except for HttpResponse_ and AggregatedHttpMessage_ which are already in a
+form of response message. You can also write your own response converter by implementing
+ResponseConverterFunction_ as follows. Also similar to RequestConverterFunction_, you can call
+``ResponseConverterFunction.fallthrough()`` when your response converter is not able to convert the result
+to an HttpResponse_.
+
+.. code-block:: java
+
+    public class MyResponseConverter implements ResponseConverterFunction {
+        @Override
+        public HttpResponse convertResponse(ServiceRequestContext ctx, Object result) {
+            if (result == MyObject.class) {
+                return HttpResponse.of(HttpStatus.OK,
+                                       MediaType.PLAIN_TEXT_UTF_8,
+                                       "Hello, %s!", result.processedName());
```

Comment:
Don't we need a code to convert the `result` to `MyObject`?

Reply:
Copy&paste mistake. `result instanceof MyObject` is correct. Let me fix it.